examples/colour-uniform/main.rs

```diff
@@ -1256,7 +1256,7 @@ impl<B: Backend> PipelineState<B> {
                 );
 
                 let shader_entries = pso::GraphicsShaderSet {
-                    vertex: vs_entry,
+                    vertex: Some(vs_entry),
```

Comment:
I think we should change the way we use this structure. The fragment stage could be moved outside, straight into the `GraphicsPipelineDesc`. And the structure with all the other stages here could be a part of the enum variant for non-mesh shaders. What do you think?

Reply:
I agree. Although I would just let shaders be freestanding inside `PrimitiveAssembler` and remove `GraphicsShaderSet`. So the final look would be:
```
pub enum PrimitiveAssembler<'a, B: Backend> {
    /// Vertex based pipeline
    Vertex {
        buffers: Vec<VertexBufferDesc>,
        attributes: Vec<AttributeDesc>,
        input_assembler: InputAssemblerDesc,
        vertex: EntryPoint<'a, B>,
        hull: Option<EntryPoint<'a, B>>,
        domain: Option<EntryPoint<'a, B>>,
        geometry: Option<EntryPoint<'a, B>>,
    },
    /// Mesh shading pipeline
    Mesh {
        task_shader: Option<EntryPoint<'a, B>>,
        mesh_shader: EntryPoint<'a, B>,
    },
}
```
Or it would be too much nesting IMHO.

Reply:
yes, that would be fine by me as well. Perhaps, the hull and domain should be merged into a tuple:
```rust
tessellation: Option<(EntryPoint, EntryPoint)>,`
```

Reply:
That's a good idea. I think this completes the API bikeshedding! :)